core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/tags/SimpleTagQueryParser.java

```diff
@@ -552,13 +565,18 @@ private void logQuery(String tenantId, Query query, String queryType, Set<? exte
     private Func1<Metric<?>, Boolean> tagValueFilter(String regexp, String tagName) {
         // If no such tagName -> no match
         return tMetric -> {
-            String tagValue = tMetric.getTags().get(tagName);
-            if(tagValue != null) {
-                boolean positive = (!regexp.startsWith("!"));
-                Pattern p = PatternUtil.filterPattern(regexp);
-                return positive == p.matcher(tagValue).matches(); // XNOR
+            try {
+                String tagValue = tMetric.getTags().get(tagName);
+                if(tagValue != null) {
+                    boolean positive = (!regexp.startsWith("!"));
+                    Pattern p = PatternUtil.filterPattern(regexp);
+                    return positive == p.matcher(tagValue).matches(); // XNOR
+                }
+                return false;
+            } catch (Exception e) {
+                logger.warn("tagValueFilter failed", e);
```

Comment:
This breaks the semantics. You will never catch the doOnError since you don't actually propagate the Exception.